### DISCUSSION TITLE
Custom Device Hangs Randomly

### DISCUSSION THREAD

**Discussion by NsinghP** [arendst/Tasmota#11828] [Q&A] [answered]
Hello, 

On the latest ESP8266-F custom device, that I have made, I am using TX GPIO-1 as an output for Relay, and I am using RX GPIO-3 as Input for a switch. I have noticed that this device hangs up every 3-4 days and I have to do Quick Power Cycle to revive it. Could this be because I am using TX and RX pins on ESP? I am mainly concerned that RX pin may be the culprit. Do Tasmota software or ESP8266-F need TX or RX pins after it has been flashed with Tasmota? On other devices I have been using TX pin but not RX pin, and these devices don't have a problem.

Regards,

Navi

**Comment by barbudor** [comment, marked as answer]
I came across recently that if TX is pulled down at boot, it prevents the ESP to start
What is your schematic on TX to Relay ?
If it is a resistor from GPIO to the base/gate of a transistor and nothing more, then that might be the problem.

**Reply by NsinghP** [reply]
On this device, wire from TX connects directly to IN pin of Relay. It is a 4 channel Relay. (I have not connected any resistor in between.)
![Four-Channel-Relay-Module-Pinout](https://user-images.githubusercontent.com/69470381/115600233-1901bc80-a2fa-11eb-815b-39b018193f46.jpg)

**Reply by barbudor** [reply]
There are obviously other parts on the board between the IN pin ad the relay.
Connecting such a board to your ESP without knowing what is on the board is just like putting your hand in a bow without knowing which animal is inside.
🦂 or 🐱

**Reply by NsinghP** [reply]
I think one of those animals is definitely biting. Schematic of IN is as follows: IN--->LED--->Optocoupler---->Resistor------>vcc. Do you think that resistor may be causing the problem?

![Screenshot (111)](https://user-images.githubusercontent.com/69470381/115604203-b9f27680-a2fe-11eb-8fb0-6e7b879960af.png)

**Reply by barbudor** [reply]
Tis diagram doesn't pulll the GPIO down, so it's unlikely to be the reason for the problem
Beside, the problem I was referring only occurs at reboot. If the unit doesn't reboot it would keep on working

Problem is somewhere else

As you are using the RX/TX for that you can't use them for long time logging Tasmota message
I would suggest to setup a syslog server and record the syslog with Syslog 3 level

**Reply by NsinghP** [reply]
Ok Thanks, I have to still figure out the syslog server setup. I have a hunch though that I may be underpowering this 4ch relay. It is being powered by 3.3v, specs recommend 3.75v to 5V. I will try to power it using JDVCC and 5v and see if that helps.

**Reply by barbudor** [reply]
The relay definitively requires 5V.
You should power the relay part through JDVCC with a strong 5V, if possible independent from the one that power you ESP.
Because of ESP being a 3V3 device, VCC should be 3V3. But in that case you don't have enough potential to power both the led and the optocoupler.
This kind of board is mainly designed to work with Arduino 5V devices.

**Reply by NsinghP** [reply]
Underpowering the Relay may explain random hangups on one other device too. On that device in addition to Relay I am using 8 extra LEDs and four TTP223 touch modules, all powered by 3.3V of AMS1117 and Sanmim 5v 700ma power supply. First I will try and see if direct 5V from power supply can operate the relay without hangs.
